File: csharp/difference-of-squares/DifferenceOfSquares.cs
using System;
using System.Linq;

public static class DifferenceOfSquares
{
    /**************************************************************************************************/
    public static int CalculateSquareOfSum(int max)
    {
        int sum = Enumerable.Range(1, max)
                            .Sum();

        return (int)(Math.Pow(sum, 2));
    }


    /**************************************************************************************************/
    public static int CalculateSumOfSquares(int max)
    {
        return Enumerable.Range(1, max)
                         .Select(s => (int)(Math.Pow(s, 2)))
                         .Sum();
    }


    /**************************************************************************************************/
    public static int CalculateDifferenceOfSquares(int max)
    {
        return CalculateSquareOfSum(max) - CalculateSumOfSquares(max);
    }
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

import "math"

func SquareOfSum(max int) int {

	var sum int

	for i := 1; i <= max; i++ {
		sum += i
	}

	return powTwoInt(sum)
}

func SumOfSquares(max int) int {
	var sum int

	for i := 1; i <= max; i++ {
		sum += powTwoInt(i)
	}

	return sum
}

func Difference(max int) int {
	return SquareOfSum(max) - SumOfSquares(max)
}

func powTwoInt(n int) int {
	return int(math.Pow(float64(n), 2))
}
